### PR TITLE
feat: add new challenge

### PR DESCRIPTION
**Pull request by Tristan22400**
Add update proxy challenge

### CODE REVIEW

File: src/ProxyDeploiment.sol
Comment (MerlinEgalite) [inline src/ProxyDeploiment.sol:1]: deployment for the name of the contract

File: src/ProxyDeploiment.sol [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+pragma solidity ^0.8.0;
+
+// These contracts are examples of contracts with bugs and vulnerabilities to practice your hacking skills.
+// DO NOT USE THEM OR GET INSPIRATION FROM THEM TO MAKE CODE USED IN PRODUCTION.
+// You are required to find vulnerabilities and problem in the contract.
+// The goal of this challenge is to update the Proxy contract from the TokenV1 to the TokenV2 contract.
+// You have on this file three contracts with one proxy contract and two implementation contracts. The owner of these contracts wants to change the implementation of his proxy from TokenV1 to TokenV2.
+// The purpose of this exercise is to give the method to change the implementation of the proxy and to find the different errors. You can reorganize the new implementation if needed.
+
```
Comment (MerlinEgalite) [inline src/ProxyDeploiment.sol:12]: space can be removed

File: src/ProxyDeploiment.sol [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+contract Proxy {
+    address internal admin;
+    address internal implementation;
+
+    constructor() {
+        admin = msg.sender;
+    }
+    /**
+     * @dev Delegates the current call to `implementation`.
```
Comment (MerlinEgalite) [inline src/ProxyDeploiment.sol:21]: NATSPECs can be cleaned

**Review by MerlinEgalite — CHANGES_REQUESTED** [review]

**Review by MerlinEgalite — CHANGES_REQUESTED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Comment by makcandrov** [comment]
Perhaps we can use `ERC20` and `Ownable` from OZ also here

**Comment by Tristan22400** [comment]
> Perhaps we can use `ERC20` and `Ownable` from OZ also here

I think it add some complexity on top of the code which is not necessary.

**Comment by Tristan22400** [comment]
What do you think @makcandrov ?

**Pull request merged**